core/src/main/scala/sttp/tapir/server/interceptor/reject/RejectHandler.scala

```diff
@@ -0,0 +1,46 @@
+package sttp.tapir.server.interceptor.reject
+
+import sttp.model.StatusCode
+import sttp.tapir._
+import sttp.tapir.server.interceptor.reject.RejectHandler._
+import sttp.tapir.server.interceptor.{RequestResult, ValuedEndpointOutput}
+
+trait RejectHandler {
```

Comment:
Looks good to me. I think you can also add a simple spec for custom `RejectHandler` in `ServerInterpreterTest`